miio/airconditioner_miot.py

```diff
@@ -0,0 +1,403 @@
+import enum
+import logging
+from typing import Any, Dict
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+_MAPPING = {
+    # Source http://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-conditioner:0000A004:xiaomi-mc4:1
+    # Air Conditioner (siid=2)
+    "power": {"siid": 2, "piid": 1},
+    "mode": {"siid": 2, "piid": 2},
+    "target_temperature": {"siid": 2, "piid": 4},
+    "eco": {"siid": 2, "piid": 7},
+    "heater": {"siid": 2, "piid": 9},
+    "dryer": {"siid": 2, "piid": 10},
+    "sleep_mode": {"siid": 2, "piid": 11},
+    # Fan Control (siid=3)
+    "fan_level": {"siid": 3, "piid": 2},
+    "vertical_swing": {"siid": 3, "piid": 4},
+    # Environment (siid=4)
+    "temperature": {"siid": 4, "piid": 7},
+    # Alarm (siid=5)
+    "buzzer": {"siid": 5, "piid": 1},
+    # Indicator Light (siid=6)
+    "led": {"siid": 6, "piid": 1},
+    # Electricity (siid=8)
+    "electricity": {"siid": 8, "piid": 1},
+    # Maintenance (siid=9)
+    "clean": {"siid": 9, "piid": 1},
+    "running_duration": {"siid": 9, "piid": 5},
+    # Enhance (siid=10)
+    "fan_percent": {"siid": 10, "piid": 1},
+    "timer": {"siid": 10, "piid": 3},
+}
+
+
+class AirConditionerMiotException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    Cool = 2
+    Dry = 3
+    Fan = 4
+    Heat = 5
+
+
+class FanLevel(enum.Enum):
+    Auto = 0
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+    Level4 = 4
+    Level5 = 5
+    Level6 = 6
+    Level7 = 7
+
+
+class AirConditionerMiotStatus:
+    """Container for status reports from the air conditioner which uses MIoT protocol."""
```

Comment:
Could you add an example response to the class docstring?

Reply:
Well, it's a MIoT device and I cannot find a example response to refer to.
Should I add something like this?
```
{
	'did': 'power',
	'siid': 2,
	'piid': 1,
	'code': 0,
	'value': False
}, {
	'did': 'mode',
	'siid': 2,
	'piid': 2,
	'code': 0,
	'value': 4
}, {
	'did': 'target_temperature',
	'siid': 2,
	'piid': 4,
	'code': 0,
	'value': 26.0
}, {
	'did': 'eco',
	'siid': 2,
	'piid': 7,
	'code': 0,
	'value': False
}, {
	'did': 'heater',
	'siid': 2,
	'piid': 9,
	'code': 0,
	'value': True
}, {
	'did': 'dryer',
	'siid': 2,
	'piid': 10,
	'code': 0,
	'value': False
}, {
	'did': 'sleep_mode',
	'siid': 2,
	'piid': 11,
	'code': 0,
	'value': False
}, {
	'did': 'fan_level',
	'siid': 3,
	'piid': 2,
	'code': 0,
	'value': 0
}, {
	'did': 'vertical_swing',
	'siid': 3,
	'piid': 4,
	'code': 0,
	'value': True
}, {
	'did': 'temperature',
	'siid': 4,
	'piid': 7,
	'code': 0,
	'value': 27.6
}, {
	'did': 'buzzer',
	'siid': 5,
	'piid': 1,
	'code': 0,
	'value': True
}, {
	'did': 'led',
	'siid': 6,
	'piid': 1,
	'code': 0,
	'value': True
}, {
	'did': 'electricity',
	'siid': 8,
	'piid': 1,
	'code': 0,
	'value': 0.0
}, {
	'did': 'clean',
	'siid': 9,
	'piid': 1,
	'code': 0,
	'value': '0,0,1,1'
}, {
	'did': 'running_duration',
	'siid': 9,
	'piid': 5,
	'code': 0,
	'value': 210.4
}
```

Reply:
That's quite verbose, but I suppose it's fine. After all, the idea is that the data structure should also be understandable for someone with no access to the device. You can use the `--debug` flag for miiocli to get the data that gets returned by the device, and just format it.